rust/examples/recordbatch-writer.rs

```diff
@@ -0,0 +1,248 @@
+/*
+ * This file is a self-contained example of writing to a Delta table using the Arrow
+ * `RecordBatch` API rather than pushing the data through a JSON intermediary
+ *
+ *
+ * This example was originally posted by @rtyler in:
+ *      <https://github.com/buoyant-data/demo-recordbatch-writer>
+ */
+
+use chrono::prelude::*;
+use deltalake::action::*;
+use deltalake::arrow::array::*;
+use deltalake::arrow::record_batch::RecordBatch;
+use deltalake::writer::{DeltaWriter, RecordBatchWriter};
+use deltalake::*;
+use log::*;
+
+use std::collections::HashMap;
+use std::path::Path;
+use std::sync::Arc;
+
+/*
+ * The main function gets everything started, but does not contain any meaningful
+ * example code for writing to Delta tables
+ */
+#[tokio::main]
+async fn main() -> Result<(), anyhow::Error> {
+    info!("Logger initialized");
+
+    let table_uri = std::env::var("TABLE_URI")?;
+    info!("Using the location of: {:?}", table_uri);
+
+    let table_path = Path::new(&table_uri);
+
+    let mut table = match Path::join(table_path, "_delta_log").is_dir() {
+        true => {
+            /* The table has been created already */
+            info!("Opening the table for writing");
+            deltalake::open_table(
+                table_path
+                    .to_str()
+                    .expect("Could not convert table path to a str"),
+            )
+            .await?
+        }
+        false => {
+            /* The table directory has not been initialized as a Delta table */
+            info!("It doesn't look like our delta table has been created");
+            create_initialized_table(&table_path).await
+        }
+    };
```

Comment:
I feel like we ought to have a better idiom for checking if a table exists, that doesn't involve introspecting the directory structure. Could we instead suggest checking for a `DeltaTableError::NotATable`? If that doesn't work, we should create an issue to improve this UX.

```suggestion
    let table = deltalake::open_table(table_path).await?;
    let mut table = {
        Ok(table) => table,
        Err(DeltaTableError::NotAtable(_)) => {
            info!("It doesn't look like our delta table has been created");
            create_initialized_table(&table_path).await
        },
        Err(err) => {
            err.unwra
        }
    }
```

Reply:
The internal object store has a is_delta_location method. Usually it should do something similar, but due to an issue with windows we actually list, which is nit the best. Nevertheless, we could expose that on the table and improve internally.